src/server/editorServices.ts

```diff
@@ -463,17 +466,33 @@ namespace ts.server {
             project.updateGraph();
         }
 
-        setCompilerOptionsForInferredProjects(projectCompilerOptions: protocol.ExternalProjectCompilerOptions): void {
-            this.compilerOptionsForInferredProjects = convertCompilerOptions(projectCompilerOptions);
+        setCompilerOptionsForInferredProjects(projectCompilerOptions: protocol.ExternalProjectCompilerOptions, projectRootPath?: string): void {
+            // ignore this settings if we are not creating inferred projects per project root.
+            if (projectRootPath && !this.useInferredProjectPerProjectRoot) return;
+
+            const compilerOptionsForInferredProjects = convertCompilerOptions(projectCompilerOptions);

```

Comment:
This local variable isn't accurately named any more since it might *not* be assigned to `this.compilerOptionsForInferredProjects`.